GVRf/Framework/jni/objects/mesh.cpp

```diff
@@ -180,12 +180,11 @@ void Mesh::getTransformedBoundingBoxInfo(glm::mat4 *Mat,
 void Mesh::generateVAO() {
 #if _GVRF_USE_GLES3_
 
-    if (vaoInitiliased_)
-        return;
+
     GLuint tmpID;
 
-    if (vao_dirty_) {
-         deleteVaos();
+    if (!vao_dirty_) {
+         return;
```

Comment:
Good catch.

The old VAO should be deleted and a new VAO created. I missed the delete call. I will add it back and ensure it is properly called. I will call deleteVaos() but test.

From: Mihail [mailto:notifications@github.com]
Sent: Thursday, October 08, 2015 9:50 AM
To: Samsung/GearVRf
Cc: Vittalbabu Srinivasan
Subject: Re: [GearVRf] Cleanup of mesh interface -- allow access only via const references. (#259)


In GVRf/Framework/jni/objects/mesh.cpp<https://github.com/Samsung/GearVRf/pull/259#discussion_r41538179>:

>      GLuint tmpID;

>

> -    if (vao_dirty_) {

> -         deleteVaos();

> +    if (!vao_dirty_) {

> +         return;

I see. When a new VAO is generated in the dirty case, what happens with the old VAO? It doesn't need explicit cleanup? Thanks.

—
Reply to this email directly or view it on GitHub<https://github.com/Samsung/GearVRf/pull/259/files#r41538179>.